rules/integrations/google_workspace/initial_access_google_workspace_object_copied_from_external_drive_and_access_granted_to_custom_application.toml

```
@@ -0,0 +1,129 @@
+[metadata]
+creation_date = "2023/03/07"
+integration = ["google_workspace"]
+maturity = "production"
+min_stack_comments = "Breaking changes for Google Workspace integration."
+min_stack_version = "8.4.0"
+updated_date = "2023/03/30"
+
+[rule]
+author = ["Elastic"]
+description = """
+Detects when a user copies a Google spreadsheet, form, document or script from an external drive. Sequence logic has
+been added to also detect when a user grants a custom Google application permission via OAuth shortly after. An
+adversary may send a phishing email to the victim with a Drive object link where "copy" is included in the URI, thus
+copying the object to the victim's drive. If a container-bound script exists within the object, execution will require
+permission access via OAuth in which the user has to accept.
+"""
+false_positives = [
+    """
+    Google Workspace users typically share Drive resources with a shareable link where parameters are edited to indicate
+    when it is viewable or editable by the intended recipient. It is uncommon for a user in an organization to manually
+    copy a Drive object from an external drive to their corporate drive. This may happen where users find a useful
+    spreadsheet in a public drive, for example, and replicate it to their Drive.
+    """,
+]
+from = "now-130m"
+index = ["filebeat-*", "logs-google_workspace*"]
+interval = "10m"
+language = "eql"
+license = "Elastic License v2"
+name = "Google Workspace Object Copied from External Drive and Access Granted to Custom Application"
+note = """## Triage and analysis
+
+### Investigating Google Workspace Resource Copied from External Drive and Access Granted to Custom Application
+
+Google Workspace user's have the capability to share access to Drive objects such as documents, sheets, forms and more. Delivery of shared access can be done by email delivery or via shared link. Shared link URIs have `view` or `edit` in the parameters based on set permissions. The `copy` parameter is another option which allows the recipient to copy the object to their own Drive, thus the object has the privileges of the recipient. Specific objects allow container-bound scripts that leverage Google's Apps Script platform. Container-bound scripts can contain malicious code that executes with the recipient's privileges if in their Drive.
+
+This rule detections when a user copies an external Drive object to their Drive storage, followed by granting permissions to a custom application via OAuth prompt.
```

```suggestion
Google Workspace users can share access to Drive objects such as documents, sheets, and forms via email delivery or a shared link. Shared link URIs have parameters like `view` or `edit` to indicate the recipient's permissions. The `copy` parameter allows the recipient to copy the object to their own Drive, which grants the object with the same privileges as the recipient. Specific objects in Google Drive allow container-bound scripts that run on Google's Apps Script platform. Container-bound scripts can contain malicious code that executes with the recipient's privileges if in their Drive.

This rule aims to detect when a user copies an external Drive object to their Drive storage and then grants permissions to a custom application via OAuth prompt.
```
Just edited for clarity and readability, not all necessary changes but suggested.

---

Thanks Isai!